### PR TITLE
feat(query-generator): add @@ operator (#1580)

### LINKED ISSUE #1580: Fulltext support
Adding support for this would be useful as it's becoming quite popular as a simpler alternative to systems like Solr, Sphinx and ElasticSearch.

Here are the docs for text search:
http://www.postgresql.org/docs/9.3/static/textsearch.html

Also, I don't know how useful this is, but here is a nice Ruby gem for Postgres full text which may help with laying out a plan on how to do it with sequelize.
https://github.com/Casecommons/pg_search

### PR DESCRIPTION
**Pull request by longuniquename**
Closes #1580

Adds `@@` text search operator for PG.
Usage:
```js
{
  where: { vector: { $ts: sequelize.fn('to_tsquery', 'test') } }
}
```
Generates:
```sql
WHERE "search"."vector" @@ to_tsquery('test')
```

### CODE REVIEW

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -2018,7 +2019,8 @@ const QueryGenerator = {
       'overlap': '$overlap',
       '&&': '$overlap',
       '@>': '$contains',
-      '<@': '$contained'
```
Comment (janmeier) [inline lib/dialects/abstract/query-generator.js]: This map is only for backwards compat, no need to add new keys to it

File: lib/model.js [hunk 2]
```diff
@@ -1408,7 +1408,7 @@ class Model {
    * ```sql
    * WHERE attr1 > 50 AND attr2 <= 45 AND attr3 IN (1,2,3) AND attr4 != 5
    * ```
-   * Possible options are: `$ne, $in, $not, $notIn, $gte, $gt, $lte, $lt, $like, $ilike/$iLike, $notLike, $notILike, $regexp, $notRegexp, '..'/$between, '!..'/$notBetween, '&&'/$overlap, '@>'/$contains, '<@'/$contained`
+   * Possible options are: `$ne, $in, $not, $notIn, $gte, $gt, $lte, $lt, $like, $ilike/$iLike, $notLike, $notILike, $regexp, $notRegexp, '..'/$between, '!..'/$notBetween, '&&'/$overlap, '@>'/$contains, '<@'/$contained, '@@'/$ts`
```
Comment (janmeier) [inline lib/model.js]: Maybe this should just refer to docs/querying :)

File: test/unit/sql/where.test.js [hunk 3]
```diff
@@ -634,6 +639,20 @@ suite(Support.getTestDialectTeaser('SQL'), () => {
             postgres: "\"userId\" NOT ILIKE ALL (ARRAY['foo','bar','baz'])"
           });
         });
+
+        suite('$ts', () => {
+          testsql('muscles', {
+            $ts: 'test'
```
Comment (janmeier) [inline test/unit/sql/where.test.js]: Add a test using sequelize.fn to mimic the real usecase

File: test/unit/sql/where.test.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-const Support   = require(__dirname + '/../support'),
-  DataTypes = require(__dirname + '/../../../lib/data-types'),
+const Support = require(__dirname+'/../support'),
+  DataTypes = require(__dirname+'/../../../lib/data-types'),
```
Comment (eseliger) [inline test/unit/sql/where.test.js]: please readd the spaces, this change is not related to this feature and it's inconsistent with the other parts of the codebase

**Comment by alikhil** [comment]
Hello @longuniquename ! Are you going to fix this PR? I am asking because this feature is very important for our team. And we are deciding should we wait for merging the PR into master or not.

**Comment by jameskennethrobinson** [comment]
Would love to see this feature merged!

**Comment by yrong** [comment]
how to pass regconfig in this solution?  maybe [pr](https://github.com/sequelize/sequelize/pull/8590/)  better
